bot/utils/persist.py

```diff
@@ -0,0 +1,68 @@
+import sqlite3
+from pathlib import Path
+from shutil import copyfile
+
+from bot.seasons.season import get_seasons
+
+DIRECTORY = Path("data")  # directory that has a persistent volume mapped to it
+
+
+def make_persistent(file_path: Path) -> Path:
+    """
+    Copy datafile at the provided file_path to the persistent data directory.
+
+    A persistent data file is needed by some features in order to not lose data
+    after bot rebuilds.
+
+    This function will ensure that a clean data file with default schema,
+    structure or data is copied over to the persistent volume before returning
+    the path to this new persistent version of the file.
+
+    If the persistent file already exists, it won't be overwritten with the
+    clean default file, just returning the Path instead to the existing file.
+
+    Note: Avoid using the same file name as other features in the same seasons
+    as otherwise only one datafile can be persistent and will be returned for
+    both cases.
+
+    Example Usage:
+    >>> import json
+    >>> template_datafile = Path("bot", "resources", "evergreen", "myfile.json")
+    >>> path_to_persistent_file = make_persistent(template_datafile)
+    >>> print(path_to_persistent_file)
+    data/evergreen/myfile.json
+    >>> with path_to_persistent_file.open("w+") as f:
+    >>>     data = json.load(f)
+    """
+    # ensure the persistent data directory exists
+    if not DIRECTORY.exists():
```

Comment:
Can skip this step by doing `DIRECTORY.mkdir(exist_ok=True)`

Reply:
TIL of a new kwarg, nice. Cheers for mentioning, this is cleaner.